R/reportDIETER.R

```diff
@@ -0,0 +1,284 @@
+    
+    names(rep_Tech) <- c("gdxfile", "model","year", "country","variable", "tech", "value")
+    
+    out_t <- rep_Tech %>% 
+      dplyr::select_(~model, ~year, ~tech,~variable, ~country,~value) %>%
+      dplyr::group_by_(~model, ~tech, ~variable, ~country) %>%
+      dplyr::mutate_(year = ~as.numeric(year)) %>%
+      complete(year = c(2010,2015,2020,2025,2030,2035,2040,2045,2050,2055,2060,2070,2080,2090,2100)) %>%
+      dplyr::mutate_all(coalesce, 0)%>%
+      dplyr::ungroup(c('model', 'tech', 'variable','country','year')) %>% 
+      dplyr::mutate_(Model = ~model, Scenario = ~paste0("baseline"), 
+             Region = ~country, Year = ~year, Value = ~round(value, digits = 4), 
+             Tech = ~tech,
+             Variable = ~variable,
+             Period = ~paste0("annual")
+      ) %>%
+      dplyr::arrange_(~Year) %>%
+      dplyr::select_(~Model,~Scenario,~Region,~Variable, ~Year,~Period, ~Tech, ~Value)
+    
+    #################################################################
+    outAnnual <- rbind(outAnnual, out)
+    outAnnual <- rbind(outAnnual, out_t)
+    
+    #Get Unit column from Variable column
+    
+    outAnnual$Unit<- substring(outAnnual$Variable, regexpr("\\(", outAnnual$Variable))
+    outAnnual <-  outAnnual %>% dplyr::mutate_(Unit =~if_else(grepl("\\(",Unit),Unit,"NA") )
+    
+    outAnnual$Variable <- mapply(gsub, "\\(.*", "",outAnnual$Variable)
+    outAnnual$Unit <- mapply(gsub, "\\(", "",  outAnnual$Unit)
+    outAnnual$Unit <- mapply(gsub, "\\)", "",  outAnnual$Unit)
+    
+    
+    return(outAnnual)
+    
+  }
+  
+  output_folder <- outputDir
+  outAnnual <- gdxToQuitteAnnual(dieterDatafile)
+  annual_new <- data.table::dcast(outAnnual, ... ~ Year, value.var="Value")
+  
+  
+  # check if these years are already present, if not then add
+  if(!'2005' %in% colnames(annual_new)){
+    annual_new$`2005` <- NA
+  }
+  
+  if(!'2110' %in% colnames(annual_new)){
+    annual_new$`2110` <- NA
+  }
+  if(!'2130' %in% colnames(annual_new)){
+    annual_new$`2130`<- NA
+  }
+  if(!'2150' %in% colnames(annual_new)){
+    annual_new$`2150` <- NA
+  }
+  
+  
+  #technology mapping
+  dieter.tech.mapping <- c(CCGT = "CCGT",
+                           #lig = "Lignite",
+                           Solar = "Solar",
+                           Wind_on = "Wind",
+                           bio = "Biomass",
+                           OCGT_eff = "OCGT",
+                           ror = "Hydro",
+                           nuc = "Nuclear"
+                           #hc = "Hard coal",
+                           #coal = "Coal (Lig + HC)",
+  )
+  
+  
+  
+  dieter.tech.mapping.standalone <- c(dieter.tech.mapping,
+                                      lig = "Lignite",
+                                      hc = "Hard coal",
+                                      coal = "Coal (Lig + HC)",
+                                      NULL)
```

Comment:
This NULL is ignored by `c()`
```suggestion
                                      )
```